### PR TITLE
Add Form Widget

### PR DESCRIPTION
**Pull request by GordonSmith**
Needed for GH-329

Signed-off-by: Gordon Smith <gordon.smith@lexisnexis.com>

### CODE REVIEW

**Comment by GordonSmith** [comment]
@jbrundage Please Review

File: src/form/Input.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        this._tag = "div";
+    }
+    Input.prototype = Object.create(HTMLWidget.prototype);
+    Input.prototype._class += " form_Input";
+    Input.prototype.implements(IInput.prototype);
+
+    Input.prototype.publish("type", "text", "set", "Input type", ["textbox", "number", "checkbox", "button", "select", "textarea"]);
+    Input.prototype.publish("selectOptions", [], "array", "Array of options used to fill a dropdown list");
+
+    Input.prototype.testData = function (domNode, element) {
```
Comment (mlzummo) [inline src/form/Input.js]: why domNode, element in the func declaration?

**Comment by GordonSmith** [comment]
@jbrundage @mlzummo Please Review

**Comment by mlzummo** [comment]
looks like there is merge conflicts now, but other than that looked good.... had that 1 question about testData ...

**Pull request merged**